src/InterpLib/Error.ml

```diff
@@ -56,6 +56,12 @@ let report ?pos ~cls msg =
   | None, _ ->
     Printf.eprintf "%s: %s\n" name msg
 
+let report_impl = ref report_to_stderr
```

Comment:
I'm not a fan of mutable state. Do we need to dynamically change this? Perhaps I'll answer this myself as I read further.

Reply:
I don't think anyone here is. However, without any mutable state we would have to pass a `report` argument to every type-checker function which seems impractical. Regular DBL code has no reason to change this variable and the language server only does it before running the type-checker, so I think there is little risk with using a ref here.